pkg/operatorloglevel/loglevel_controller.go

```diff
@@ -0,0 +1,107 @@
+package operatorloglevel
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+)
+
+type operatorLogLevelNormalizer struct {
+	dynamicClient dynamic.Interface
+}
+
+// NewLogLevelNormalizer normalizes the log level in operator custom resource
+// if it's value differs from supported values
+func NewLogLevelNormalizer(dynamicClient dynamic.Interface,
+	operatorClient operatorv1helpers.OperatorClient,
+	recorder events.Recorder) factory.Controller {
+	c := operatorLogLevelNormalizer{
+		dynamicClient: dynamicClient,
+	}
+
+	return factory.New().
+		WithSync(c.sync).
+		WithSyncDegradedOnError(operatorClient).
+		ResyncEvery(time.Minute).
+		ToController("OperatorLogLevelNormalizer", recorder)
+}
+
+// sync runs periodically and makes sure that the operator log level values does not
+// differ from supported values. Sub-resources are ignored. Operator custom resources
+// that do not have log level field are also ignored.
+func (c *operatorLogLevelNormalizer) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	// find all operator CRDs
+	apiResources, err := c.dynamicClient.Resource(schema.GroupVersionResource{
+		Group:   "operator.openshift.io",
+		Version: "v1",
+	}).List(ctx, metav1.ListOptions{})
```

Comment:
does this work? I think you have to pass a concrete resource name.

There is the discovery client to get the list.

Reply:
Maybe this work because the dynamic client hits the right endpoint. But better use an explicit discovery client. This here is only an accident that it works ;-)

Reply:
Actually, after reading https://github.com/openshift/api/pull/736, I would just put an hard-coded slice of GVRs here. No discovery. Then also know that they are really using the underlying type with log levels we expect.

Reply:
> This here is only an accident that it works ;-)
I tested the changes yesterday and they work. I was just lucky I guess :)

>  I would just put an hard-coded slice of GVRs here
hard-coded the gvr slice now. No discovery and dynamic client.